src/tools/wasm-reduce.cpp

```diff
@@ -842,16 +838,36 @@ struct Reducer
     if (it != module->elementSegments.end()) {
       first = it->get()->data[0];
     }
-
     // try to reduce to first function. first, shrink segment elements.
     // while we are shrinking successfully, keep going exponentially.
     bool shrank = false;
     for (auto& segment : module->elementSegments) {
       shrank = shrinkByReduction(segment.get(), 100);
     }
     // the "opposite" of shrinking: copy a 'zero' element
+    if (first == nullptr) {
+      return;
+    }
```

Comment:
this if could be on line 840

but i don't follow this, lower down it checks if `!first` which is no longer possible?

Reply:
The Alpine build was failing on this even though I had this check before entering the loop. (I suspect it had something to do with `first` being used in a lambda.)